src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -2,67 +2,91 @@
 
 import org.skypro.skyshop.product.Product;
 
-public class ProductBasket {
-    private static int maxProduct = 5;
-    private Product[] basket = new Product[maxProduct];
+import java.util.*;
 
+public class ProductBasket {
+    private HashMap<String, ArrayList<Product>> basket = new HashMap();
```

Comment:
При объявлении переменных стараемся использовать максимально возможный уровень абстракции. То есть объявлять не HashMap, а Map

```suggestion
    private Map<String, ArrayList<Product>> basket = new HashMap();
```